geomstats/distributions/lognormal.py

```diff
@@ -0,0 +1,187 @@
+        is defined by transforming the mean -- an SPD matrix -- into
+        a symmetric matrix through the Matrix logarithm, which gives
+        the element "log-mean" that now belongs to a vector space.
+        This log-mean and given cov are used to parametrize a Normal
+        Distribution.
+
+        b)AffineInvariant Metric : X is distribution as LogNormal(mean, cov)
+        if exp(mean^{1/2}.X.mean^{1/2}) is distributed as Normal(0, cov)
+
+    Parameters
+    ----------
+    manifold : Manifold obj, {Euclidean(n), SPDMatrices(n)}
+        Manifold to sample over. Manifold should
+        be instance of Euclidean or SPDMatrices.
+    mean : array-like,
+            shape=[dim] if manifold is Euclidean Space
+            shape=[n, n] if manifold is SPD Manifold
+        Mean of the distribution.
+    cov : array-like,
+            shape=[dim, dim] if manifold is Euclidean Space
+            shape=[n*(n+1)/2, n*(n+1)/2] if manifold is SPD Manifold
+        Covariance of the distribution.
+
+    Returns
+    --------
+    samples : array-like, shape=[..., dim]
+        Samples from LogNormal distribution.
+
+    Example
+    --------
+    >>> import geomstats.backend as gs
+    >>> from geomstats.geometry.spd_matrices import SPDMatrices
+    >>> from geomstats.distributions.lognormal import LogNormal
+    >>> mean = 2 * gs.eye(3)
+    >>> cov  = gs.eye(6)
+    >>> SPDManifold = SPDMatrices(3)
+    >>> LogNormalSampler = LogNormal(SPDManifold, mean, cov)
+    >>> data = LogNormalSampler.sample(5)
+
+    References
+    ----------
+    .. [LNGASPD2016] A. Schwartzman,
+    "LogNormal distributions and"
+    "Geometric Averages of Symmetric Positive Definite Matrices.",
+    International Statistical Review 84.3 (2016): 456-486.
+    """
+    def __init__(self, manifold, mean, cov=None):
+
+        if (
+            not isinstance(manifold, SPDMatrices) and
+            not isinstance(manifold, Euclidean)
+        ):
+            raise ValueError(
+                "Invalid Manifold object, "
+                "Should be of type SPDMatrices or Euclidean")
+
+        if not manifold.belongs(mean):
+            raise ValueError(
+                "Invalid Value in mean, doesn't belong to ",
+                type(manifold).__name__)
+
+        n = mean.shape[-1]
+        metric = manifold.metric
+        if isinstance(manifold, Euclidean):
+            cov_n = n
+            if metric is None:
+                manifold.metric = EuclideanMetric(n)
+            else:
+                metric = manifold.metric
+                if not isinstance(metric, EuclideanMetric):
+                    raise ValueError(
+                        "Invalid Metric, "
+                        "Should be of type EuclideanMetric")
+
+        else:
+            cov_n = (n * (n + 1)) // 2
+            if metric is None:
+                manifold.metric = SPDMetricLogEuclidean(n)
+            else:
+                metric = manifold.metric
```

Comment:
I'm worried that the `.metric` attribute of the parameter passed as input will be modified too. Maybe create a `self.metric` for this class and store the metric there?

Reply:
Now, user cannot change `manifold`, `mean`, `cov`  or `metric` after initializing !!